sdk/cognitiveservices/azure-cognitiveservices-formrecognizer/DESIGN_DOC.md

```diff
@@ -0,0 +1,544 @@
+    source: str, source_prefix_filter: str, include_sub_folders: bool=False
+) -> LROPoller -> CustomLabeledModel
+
+client.begin_training(
+    source: str, source_prefix_filter: str, include_sub_folders: bool=False
+) -> LROPoller -> CustomModel
+
+# Extract
+client.begin_extract_forms(form: Union[str, BytesIO], model_id: str) -> LROPoller -> List[ExtractedPage]
+
+client.begin_extract_labeled_forms(form: Union[str, BytesIO], model_id: str) -> LROPoller -> ExtractedForm
+
+# Manage
+client.list_custom_models() -> ItemPaged[ModelInfo]
+
+client.get_models_summary() -> ModelsSummary
+
+client.get_custom_model(model_id: str) -> CustomModel
+
+client.get_custom_labeled_model(model_id: str) -> CustomLabeledModel
+
+client.delete_custom_model(model_id: str) -> None
+```
+
+### Custom Models Unlabeled
+```python
+# Training ---------------------------------------------------
+class CustomModel:
+    model_id: str
+    status: str
+    created_on: ~datetime.datetime
+    last_updated_on: ~datetime.datetime
+    train_result: TrainResult
+
+class TrainResult:
+    extracted_fields: List[FormFields]
+    documents: List[TrainingDocumentInfo]
+    errors: List[FormRecognizerError]
+
+class FormFields:
+    form_type_id: int
+    fields: List[str]
+
+class TrainingDocumentInfo:
+    document_name: str
+    status: str
+    page_count: int
+    errors: List[FormRecognizerError]
+
+class FormRecognizerError:
+    code: str
+    message: str
+
+# Analyze ---------------------------------------------------
+class ExtractedPage:
+    fields: List[ExtractedField]
+    tables: List[ExtractedTable]
+    page_metadata: PageMetadata
+    page_number: int
+    form_type_id: int
+
+class ExtractedField:
+    name: ExtractedText
+    value: ExtractedText
+    confidence: float
+
+class ExtractedText:
+    text: str
+    bounding_box: List[float]
+    elements: List[Union[ExtractedLine, ExtractedWord]]
+
+class ExtractedLine:
+    text: str
+    bounding_box: List[float]
+    language: str
+    page_number: int
+    words: List[ExtractedWord]
+
+class ExtractedWord:
+    text: str
+    bounding_box: List[float]
+    confidence: float
+    page_number: int
+    line: ExtractedLine
+
+class PageMetadata:
+    page_number: int
+    angle: float
+    width: int
+    height: int
+    unit: str
+    language: str
+    lines: List[ExtractedLine]
+```
+
+### Custom Models Labeled
+```python
+# Training ---------------------------------------------------
+class CustomLabeledModel:
+    model_id: str
+    status: str
+    created_on: ~datetime.datetime
+    last_updated_on: ~datetime.datetime
+    train_result: LabeledTrainResult
+
+class LabeledTrainResult:
+    fields: List[FieldNames]
+    average_model_accuracy: float
+    documents: List[TrainingDocumentInfo]
+    errors: List[FormRecognizerError]
+
+class TrainingDocumentInfo:
+    document_name: str
+    status: str
+    page_count: int
+    errors: List[FormRecognizerError]
+
+class FieldNames:
```

Comment:
Suggest rename to `FieldName` (not plural) or maybe `FieldInfo`.

Reply:
Renamed to `FieldInfo`